Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -402,6 +403,57 @@ void GraphicsContextSkia::drawSkiaPath(const SkPath& path, SkPaint& paint)
         endTransparencyLayer();
 }
 
+bool GraphicsContextSkia::drawPathAsSingleElement(const Path& path, SkPaint& paint)
+{
+    if (hasDropShadow())
+        return false;
+
+    auto segment = path.singleSegment();
+    if (!segment)
+        return false;
+
+    auto drawArc = [&](const FloatPoint& center, float radius, float startAngle, float endAngle, bool isClosedArc) -> bool {
```

Comment:
You could take `const PathArc& arc` here:

```suggestion
    auto drawArc = [&](const PathArc& arc, bool isClosedArc) -> bool {
```

then you can call `drawArc(arc, true)` or `drawArc(closedArc.arc, false)`